mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -397,46 +434,118 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
 
     /** Trim the fitted value so that it will be in valid range. */
     def project(mu: Double): Double = mu
+
   }
 
   private[regression] object Family {
 
     /**
-     * Gets the [[Family]] object from its name.
+     * Gets the [[Family]] object based on family and variancePower.
+     * 1) retrieve object based on family name
+     * 2) if family name is tweedie, retrieve object based on variancePower
      *
-     * @param name family name: "gaussian", "binomial", "poisson" or "gamma".
+     * @param model a GenerealizedLinearRegressionBase object
      */
-    def fromName(name: String): Family = {
-      name match {
-        case Gaussian.name => Gaussian
-        case Binomial.name => Binomial
-        case Poisson.name => Poisson
-        case Gamma.name => Gamma
+    def fromModel(model: GeneralizedLinearRegressionBase): Family = {
```

Comment:
Rename to ```fromParams```, we extract ```family``` and ```variancePower``` from the ```Params``` which is the superclass of GLR estimator and model. And actually we use this function for both estimator(L279) and model(L974).